### PR TITLE
Add inactivity state and fix hostname bug

### PR DESCRIPTION
**Pull request by bluelhf**
This pull request adds inactivity and fixes the issues with hostnames. When no carets are moved within 30 seconds, the plugin goes into an inactive state, and won't send activity payloads. Additionally, the plugin will now correctly determine the hostname using OS-specific tooling instead of using the DNS configuration of the machine.

### CODE REVIEW

File: src/main/kotlin/fi/testaustime/plugin_intellij/services/TestaustimeProjectService.kt [hunk 1]
```diff
@@ -1,32 +1,45 @@
...
 
+    fun isActive(): Boolean {
+        return caretListener.isActive()
+    }
+
     companion object {
         var projects: MutableList<Project> = ArrayList();
         fun broadcast(tokenValid: Boolean) {
             for (project in projects) {
                 if (tokenValid) {
```
Comment (developerfromjokela) [inline src/main/kotlin/fi/testaustime/plugin_intellij/services/TestaustimeProjectService.kt:31]: Notifications could become annoying after some time. Would it be a good idea to let the user to disable them completely in settings? 😄

Comment (developerfromjokela) [reply src/main/kotlin/fi/testaustime/plugin_intellij/services/TestaustimeProjectService.kt:31]: I'm talking about notice messages, no issues with error messages.

Comment (bluelhf) [reply src/main/kotlin/fi/testaustime/plugin_intellij/services/TestaustimeProjectService.kt:31]: Yes, though the notification system was present before this PR, so it should probably be updated separately. It'll need a flag in settings state and some sort of checkbox in the Testaustime settings.

**Comment by bluelhf** [comment]
lgtm
hörönaurut

**Pull request merged**